libs/logging/src/logger.ts

```diff
@@ -1,77 +1,30 @@
-import { Dictionary } from '@votingworks/types';
-import makeDebug from 'debug';
-import { LogEventId, getDetailsForEventId } from './log_event_ids';
-import { CLIENT_SIDE_LOG_SOURCES, LogSource } from './base_types/log_source';
-import {
-  LogDisposition,
-  LogDispositionStandardTypes,
-  LogLine,
-  LoggingUserRole,
-} from './types';
+import { LogEventId } from './log_event_ids';
+import { LogSource } from './base_types/log_source';
+import { LogLine, LoggingUserRole } from './types';
+import { LogData, BaseLogger } from './base_logger';
 
-export const LOGS_ROOT_LOCATION = '/var/log';
-export const LOG_NAME = 'vx-logs';
-export const FULL_LOG_PATH = `${LOGS_ROOT_LOCATION}/${LOG_NAME}.log`;
-
-const debug = makeDebug('logger');
-
-interface LogData extends Dictionary<string | boolean | number> {
-  message?: string;
-  disposition?: LogDisposition;
-}
-
-export class Logger {
+export class Logger extends BaseLogger {
   constructor(
-    private readonly source: LogSource,
-    private readonly kiosk?: KioskBrowser.Kiosk
-  ) {}
-
-  getSource(): LogSource {
-    return this.source;
+    source: LogSource,
+    private readonly getUserRole: () => Promise<LoggingUserRole>
+  ) {
+    super(source);
   }
 
-  async log(
+  async logAsCurrentUser(
```

Comment:
Is it possible to have this method just be called `log`? I'm thinking it might be more obvious to use as the main method of the logger that way (e.g. when using auto-complete).

Reply:
I'm also thinking that maybe we don't need to have the `BaseLogger.log` method available on `Logger`

Reply:
I made it different because, to make this refactor easier, I have `Logger` extend `BaseLogger`, which allows a lot of tests to remain unchanged. But I can't change the signature of of a function in an extended class.